daemon/system-info.sh

```diff
@@ -8,111 +8,182 @@ KERNEL_VERSION="$(uname -r)"
 ARCHITECTURE="$(uname -m)"
 
 # -------------------------------------------------------------------------------------------------
-# detect the operating system
+# detect the virtualization
 
-OS_DETECTION="unknown"
-NAME="unknown"
-VERSION="unknown"
-VERSION_ID="unknown"
-ID="unknown"
-ID_LIKE="unknown"
+if [ -z ${VIRTUALIZATION} ]; then
+    VIRTUALIZATION="unknown"
+    VIRT_DETECTION="none"
 
-if [ "${KERNEL_NAME}" = "Darwin" ]; then
-	# Mac OS
-	OIFS="$IFS"
-	IFS=$'\n'
-	set $(sw_vers) > /dev/null
-	NAME=$(echo $1 | tr "\n\t" '  ' | sed -e 's/ProductName:[ ]*//' -e 's/[ ]*$//')
-	VERSION=$(echo $2 | tr "\n\t" '  ' | sed -e 's/ProductVersion:[ ]*//' -e 's/[ ]*$//')
-	ID="mac"
-	ID_LIKE="mac"
-	OS_DETECTION="sw_vers"
-	IFS="$OIFS"
+    if [ -n "$(command -v systemd-detect-virt 2>/dev/null)" ]; then
+            VIRTUALIZATION="$(systemd-detect-virt -v)"
+            VIRT_DETECTION="systemd-detect-virt"
+            CONTAINER="$(systemd-detect-virt -c)"
+            CONT_DETECTION="systemd-detect-virt"
+    else
+            if grep -q "^flags.*hypervisor" /proc/cpuinfo 2>/dev/null; then
+                    VIRTUALIZATION="hypervisor"
+                    VIRT_DETECTION="/proc/cpuinfo"
+            elif [ -n "$(command -v dmidecode)" ]; then
+                    # Virtualization detection from https://unix.stackexchange.com/questions/89714/easy-way-to-determine-virtualization-technology
+                    # This only works as root
+                    if dmidecode -s system-product-name 2>/dev/null | grep -q "VMware\|Virtual\|KVM\|Bochs"; then
```

Comment:
Please, take a look in the first answer of this link `https://unix.stackexchange.com/questions/89714/easy-way-to-determine-virtualization-technology`, I think we can add more expected results for `grep`.

Reply:
That's where I took the code from :) The other answers require parsing the output of different sources. I've added it to the description of the follow-up issue that @Ferroin is going to work on as he knows about which parts will need extra protection (such as the cases where `dmidecode` can hang a VM). I added the comment into the script as a marker.

Reply:
All right :)